### PR TITLE
Haikuporter selfhelp

### PR DESCRIPTION
**Pull request by kallisti5**
Begin breaking the massive arg list into groupings.  Some of these flags are "actions" some are basic, advanced... some only apply to buildmaster mode.

### CODE REVIEW

**Comment by kallisti5** [comment]
Before:
[before.txt](https://github.com/haikuports/haikuporter/files/7714973/before.txt)

After:
[after.txt](https://github.com/haikuports/haikuporter/files/7714977/after.txt)

**Comment by kallisti5** [comment]
Updated language around build master mode to be more consistent.

{{{
diff --git a/HaikuPorter/Options.py b/HaikuPorter/Options.py
index d609ca5..bf6e863 100644
--- a/HaikuPorter/Options.py
+++ b/HaikuPorter/Options.py
@@ -320,8 +320,8 @@ def parseOptions():
                dest='checkPortsReleases', default=False,
                help='check for newer releases of ports published upstream')
 
-       buildmaster_flags = OptionGroup(parser, "Buildmaster Flags",
-               "Options only relevent to Haikuporter running in buildmaster mode")
+       buildmaster_flags = OptionGroup(parser, "Build Master Options",
+               "Options only relevent to Haikuporter running in build master mode")
        buildmaster_flags.add_option('--system-packages-directory', action='store',
                type='string', dest='systemPackagesDirectory', default=None,
                help='specifies the directory to be used to look up system packages')
}}}

File: HaikuPorter/Options.py [hunk 1]
```diff
@@ -217,119 +241,104 @@ def parseOptions():
...
 		dest='activeVersionsOnly', default=False,
 		help='only check in active versions of ports instead of all ports')
-
-	parser.add_option('--check-ports-releases', action='store_true',
+	advanced_flags.add_option('--check-ports-releases', action='store_true',
 		dest='checkPortsReleases', default=False,
 		help='check for newer releases of ports published upstream')
 
+	buildmaster_flags = OptionGroup(parser, "Build Master Options",
+		"Options only relevent to Haikuporter running in build master mode")
```
Comment (pulkomandy) [inline HaikuPorter/Options.py:324]:
```suggestion
		"Options only relevant to Haikuporter running in build master mode")
```

Comment (kallisti5) [reply HaikuPorter/Options.py]: done

**Pull request merged**